### PR TITLE
fix(useSortable): prevent from creating multi instances

### LINKED ISSUE #3496: useSortable can start multiple instances of Sortable
### Clear and concise description of the problem

The call to useSortable can start multiple instances in cases where the target element is provided right away when the useSortable function is invoked and then the start method is called again manually.

### Suggested solution

diff --git a/node_modules/@vueuse/integrations/useSortable.mjs b/node_modules/@vueuse/integrations/useSortable.mjs
index d21f3a7..00d4ea7 100644
--- a/node_modules/@vueuse/integrations/useSortable.mjs
+++ b/node_modules/@vueuse/integrations/useSortable.mjs
@@ -12,7 +12,7 @@ function useSortable(el, list, options = {}) {
   };
   const start = () => {
     const target = typeof el === "string" ? document == null ? void 0 : document.querySelector(el) : unrefElement(el);
-    if (!target)
+    if (!target || sortable != null)
       return;
     sortable = new Sortable(target, { ...defaultOptions, ...resetOptions });
   };

### Alternative

_No response_

### Additional context

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by Doctor-wu**
### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [x] Ideally, include relevant tests that fail without this PR but pass with it.

<details>
<summary><strong>⚠️ Slowing down new functions</strong></summary>
<br>

> **Warning**: **Slowing down new functions**
>
> As the VueUse audience continues to grow, we have been inundated with an overwhelming number of feature requests and pull requests. As a result, maintaining the project has become increasingly challenging and has stretched our capacity to its limits. As such, in the near future, we may need to slow down our acceptance of new features and prioritize the stability and quality of existing functions. **Please note that new features for VueUse may not be accepted at this time.** If you have any new ideas, we suggest that you first incorporate them into your own codebase, iterate on them to suit your needs, and assess their generalizability. If you strongly believe that your ideas are beneficial to the community, you may submit a pull request along with your use cases, and we would be happy to review and discuss them. Thank you for your understanding.

</details>

---

### Description

fix #3496 
prevent start multi instance

### Additional context

---

### <samp>🤖 Generated by Copilot at 9afb592</samp>

This pull request fixes a bug in `onClickOutside` and improves the performance and reliability of `useSortable`. It ensures that the `onClickOutside` function correctly handles null or undefined target elements, and that the `useSortable` function does not create duplicate Sortable instances.

### <samp>🤖 Generated by Copilot at 9afb592</samp>

* Fix a bug in `onClickOutside` that caused incorrect `shouldListen` value when target element was null or undefined ([link](https://github.com/vueuse/vueuse/pull/3501/files?diff=unified&w=0#diff-8b308f45dba561e732d94afb100d76cc2aee8b7d265690c72e6fdf1e9dc59925L93-R93))
* Prevent creating duplicate Sortable instances in `useSortable` by checking if one already exists for the target element ([link](https://github.com/vueuse/vueuse/pull/3501/files?diff=unified&w=0#diff-fedee44a635d5e344c84f28fd3206526afd89bf85dd2176067647e21091b70abL54-R54))

### CODE REVIEW

**Comment by Doctor-wu** [comment]
@antfu I have submitted a pull request to replace the PR https://github.com/vueuse/vueuse/pull/3470.

**Comment by Alfred-Skyblue** [comment]
We should set `sortable` to `undefined` when stopping, and when starting, we should check if `sortable` exists; if it does, we should not create a new `sortable`.

**Comment by Doctor-wu** [comment]
> We should set `sortable` to `undefined` when stopping, and when starting, we should check if `sortable` exists; if it does, we should not create a new `sortable`.停止时我们应该将sortable设置为undefined，启动时我们应该检查sortable是否存在；如果是这样，我们不应该创建一个新的可排序对象。

@Alfred-Skyblue Got you, so I think we need to set the sortable to undefined when the stop function is invoked

File: packages/integrations/useSortable/index.ts [hunk 1]
```diff
@@ -51,12 +51,15 @@ export function useSortable<T>(
 
   const start = () => {
     const target = (typeof el === 'string' ? document?.querySelector(el) : unrefElement(el))
-    if (!target)
+    if (!target || sortable !== undefined)
```
Comment (Alfred-Skyblue) [inline packages/integrations/useSortable/index.ts:54]: if (!target || sortable)

Comment (Doctor-wu) [reply packages/integrations/useSortable/index.ts:54]: I think this is a code style problem 😂, the type of sortable is ```Sortable | undefined```, if sortable doesn't equal undefined, it means sortable exists.

**Review by Alfred-Skyblue — APPROVED** [review]

**Comment by brgkcosic** [comment]
does this inherently then stop from creating recursive groups that can be sorted? I need to implement a functionality that groups items inside, and this group can either render a group item or another group (recursively) - so my logic is to create groups that are sortable and eventual rendering of items when reaching the bottom line.

**Comment by Alfred-Skyblue** [comment]
@brgkcosic  If you need to perform recursive sorting, you should create a separate sortable instance at each level of recursion.

**Pull request merged**